ledger-core/v2/testutils/slotdebugger/stepmachine.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package slotdebugger
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/gojuno/minimock/v3"
+
+	"github.com/insolar/assured-ledger/ledger-core/v2/conveyor"
+	"github.com/insolar/assured-ledger/ledger-core/v2/conveyor/smachine"
+	"github.com/insolar/assured-ledger/ledger-core/v2/instrumentation/inslogger"
+	messageSenderAdapter "github.com/insolar/assured-ledger/ledger-core/v2/network/messagesender/adapter"
+	"github.com/insolar/assured-ledger/ledger-core/v2/pulse"
+	testUtilsCommon "github.com/insolar/assured-ledger/ledger-core/v2/testutils"
+	"github.com/insolar/assured-ledger/ledger-core/v2/testutils/messagesender"
+	"github.com/insolar/assured-ledger/ledger-core/v2/vanilla/longbits"
+	"github.com/insolar/assured-ledger/ledger-core/v2/vanilla/synckit"
+	"github.com/insolar/assured-ledger/ledger-core/v2/vanilla/throw"
+	"github.com/insolar/assured-ledger/ledger-core/v2/virtual/integration/convlog"
+)
+
+const (
+	runTimeout = 30 * time.Second
+)
+
+type StepController struct {
+	t   *testing.T
+	ctx context.Context
+
+	externalSignal synckit.VersionedSignal
+	internalSignal synckit.VersionedSignal
+
+	SlotMachine   *smachine.SlotMachine
+	debugLogger   *testUtilsCommon.DebugMachineLogger
+	stepIsWaiting bool
+	watchdog      *watchdog
+
+	worker        *worker
+	MessageSender *messagesender.ServiceMockWrapper
+}
+
+func New(ctx context.Context, t *testing.T, suppressLogError bool) *StepController {
+	inslogger.SetTestOutput(t, suppressLogError)
+
+	debugLogger := testUtilsCommon.NewDebugMachineLogger(convlog.MachineLogger{})
+	machineConfig := smachine.SlotMachineConfig{
+		PollingPeriod:     500 * time.Millisecond,
+		PollingTruncate:   1 * time.Millisecond,
+		SlotPageSize:      1000,
+		ScanCountLimit:    1e5,
+		LogAdapterCalls:   true,
+		SlotMachineLogger: debugLogger,
+	}
+
+	w := &StepController{
+		t:           t,
+		ctx:         ctx,
+		debugLogger: debugLogger,
+	}
+	w.SlotMachine = smachine.NewSlotMachine(machineConfig,
+		w.internalSignal.NextBroadcast,
+		combineCallbacks(w.externalSignal.NextBroadcast, w.internalSignal.NextBroadcast),
+		nil,
+	)
+	w.worker = newWorker(w)
+
+	pd := pulse.NewFirstPulsarData(10, longbits.Bits256{})
+	pulseSlot := conveyor.NewPresentPulseSlot(nil, pd.AsRange())
+	w.SlotMachine.AddDependency(&pulseSlot)
```

Comment:
Please, move that block to separate method (to increase readability of constructor